proto/spacemesh/v1/app_types.proto

```diff
@@ -0,0 +1,111 @@
+syntax = "proto3";
+package spacemesh.v1;
+option go_package = "github.com/spacemeshos/api/release/go/spacemesh/v1";
+import "spacemesh/v1/types.proto";
+import "spacemesh/v1/var_types.proto";
+
+/////////////// Template API
+
+// The address of an app template in the templates global state store
+message TemplateAddress {
+    bytes address = 1;
+}
+
+// Template deploy data
+message DeployData {
+    TemplateAddress address = 1; // template address
+    // note - the author can be obtained from the deploy transaction but it is nice to have it here
+    AccountId author = 2; // template uploader
+    LayerNumber deploy_layer = 3; // layer that the template was deployed - 0 for genesis templates
+    TransactionId deploy_tx_id = 4; // the id of the deploy template transaction. Empty for genesis templates.
+}
+
+// Template code
+message Code {
+    string target_svm_version = 4; // at deploy time
```

Comment:
Can we collapse target svm version and kind down into one field? Can't it be included in the code itself as a "magic word" (first few bytes)?